staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -149,19 +149,30 @@ func (g *genValidations) GenerateType(c *generator.Context, t *types.Type, w io.
 }
 
 func (g *genValidations) hasValidations(n *typeNode) bool {
+	seen := map[*typeNode]bool{}
+	return g.hasValidationsImpl(n, seen)
+}
+
+func (g *genValidations) hasValidationsImpl(n *typeNode, seen map[*typeNode]bool) bool {
+	if seen[n] {
+		return false
+	}
+	seen[n] = true
+
 	if n == nil {
```

Comment:
nit: it might make sense to make the nil check the first check here, fine either way though

Reply:
ACK